src/nupic/types/Serializable.hpp

```diff
@@ -142,29 +148,32 @@ typedef enum {BINARY, PORTABLE, JSON, XML} SerializableFormat;
 // them. The Cereal package does not need to be exposed in the API.
 //
 // When setting up a new class that should be serialized, do the following:
-// 1) subclass it from Serialization.
-// 2) Add CerealAdapter macro and save(Archive& ar) and load(Archive& ar) methods
+// 1) subclass it from Serializable (This class).
+// 2) Add CerealAdapter macro and save_ar(Archive& ar) and load_ar(Archive& ar) methods
 //    to the class as shown at the end of this file.
-// 3) Implement the save and load functions as described in the Cereal documentation
+// 3) Implement the save_ar and load_ar functions as described in the Cereal documentation
 //    see http://uscilab.github.io/cereal/index.html
 //    - Note that you can directly serialize:
 //      a) Any fundemental data type (numeric values).
 //      b) A std::string
 //      c) Most of the STL containers.
 //      d) Any other class that is subclassed from Serialize.
 //      e) Any smart pointer (std::shared_ptr) if contains above types, non-array.
-//    - It will not serialize pointers or arrays directly.
+//    - It will NOT serialize pointers or arrays directly.
```

Comment:
nit, can you add a comment how to overcome that?

Reply:
> - It will NOT serialize pointers or arrays directly.
nit, can you add a comment how to overcome that?

I wish I knew how to do that.  I guess I could say to put it in one of the STL containers.